### ISSUE TITLE
Standard bivariate normal distribution

### ISSUE THREAD

**Issue by rtrangucci** [stan-dev/math#962]
## Description
Add the standard bivariate normal distribution with scalar inputs. Specifically, add:

* [ ] std_binormal_lpdf(reals z1, reals z2 | reals rho)
* [ ] std_binormal_cdf(reals z1, reals z2 | reals rho)
* [ ] std_binormal_lcdf(reals z1, reals z2 | reals rho)
* [ ] std_binormal_lccdf(reals z1, reals z2 | reals rho)
* [ ] std_binormal_rng(reals rho)

## Example
Bivariate probit model
```
data {
  int<lower=1> N;
  int<lower=1> K1;
  int<lower=1> K2;
  int<lower=0,upper=1> y1[N];
  int<lower=0,upper=1> y2[N];
  matrix[N, K1] X1;
  matrix[N, K2] X2;
}
parameters {
  real<lower=-1,upper=1> rho;
  vector[K1] beta_1;
  vector[K2] beta_2;
  real alpha_1;
  real alpha_2;
}
model {
  vector[N] mu_1 = alpha_1 + X1 * beta_1;
  vector[N] mu_2 = alpha_2 + X2 * beta_2;
  for (n in 1:N) {
    if(y1[n] == 1 && y2[n] == 1) {
      target += std_binormal_lcdf(mu_1[n], mu_2[n], rho);
    }
    if(y1[n] == 0 && y2[n] == 1) {
      target += std_binormal_lcdf(-mu_1[n], mu_2[n], -rho);
    }
    if(y1[n] == 1 && y2[n] == 0) {
      target += std_binormal_lcdf(mu_1[n], -mu_2[n], -rho);
    }
    if(y1[n] == 0 && y2[n] == 0) {
      target += std_binormal_lcdf(-mu_1[n], -mu_2[n], rho);
    }
  }
}
```

## Additional Information
binormal_cdf and binormal_lcdf can implement the following algorithms:

* http://www.math.wsu.edu/faculty/genz/papers/bvnt.pdf
* https://arxiv.org/pdf/1004.3616.pdf
* https://papers.ssrn.com/sol3/papers.cfm?abstract_id=2924071
* https://projecteuclid.org/euclid.aos/1176344739

## Current Math Version
v2.18.0

**Comment by bgoodri** [comment]
My guess is the Pan paper is the best approach for the `_lcdf` but we would have to have a heuristic for how many terms of the infinite sum to compute based on `Phi(a)`, `Phi(b)`, `asin(rho)`, etc.

**Comment by rtrangucci** [comment]
Yeah, that sounds good to me. Here's a reference for the gradient of the CDF:

**Comment by rtrangucci** [comment]
FYI, I'm waiting on #978 to work on the nonstandardized distributions.
